crates/uv/src/commands/tool/upgrade.rs

```diff
@@ -156,6 +206,13 @@ async fn upgrade_tool(
         }
     };
 
+    // If a new Python version was requested for this package, create a new environment
+    if let (Some(python_request), Some(interpreter)) = (python_request, interpreter) {
+        if !python_request.satisfied(environment.interpreter(), cache) {
```

Comment:
Lets make this stricter. Check out the logic we added to `tool/install.rs` recently (grep for `let same_interpreter = if cfg!(windows) { ... }`).

Reply:
Awesome, done — I also factored this check into a method on ```Interpreter``` in line with one of Zanie's todos (went ahead and removed it, hope that's okay).